docs/data-sources/ec_trafficfilter.md

```diff
@@ -0,0 +1,34 @@
+---
+page_title: "Elastic Cloud: ec_trafficfilter"
+description: |-
+  Filters for available traffic filters that match the given criteria
+---
+
+# Data Source: ec_trafficfilter
+
+Use this data source to filter for an existing traffic filter that has been created via one of the provided filters. 
+
+## Example Usage
+
+```hcl
+data "ec_trafficfilter" "name" {
+  name = "example-filter"
+}
+
+data "ec_trafficfilter" "id" {
+  id = "41d275439f884ce89359039e53eac516"
+}
+
+data "ec_trafficfilter" "region" {
+  region = "us-east-1"
+}
+```
+
+## Argument Reference
+
+* `name` (Optional) - The name of the traffic filter. Has to match exactly.
+* `id` (Optional) - The id of the traffic filter. Has to match exactly.
+* `region` (Optional) - Region where the traffic filter is. For Elastic Cloud Enterprise (ECE) installations, use `"ece-region`.
+
+## Attributes Reference
+See https://www.elastic.co/guide/en/cloud/current/definitions.html#TrafficFilterRulesets for the API guide
```

Comment:
I deliberately didn't document the actual structure here because I felt that wasn't the right choice. Why am I copy/pasting the structure from the link and writing up my own descriptions (or copying them) when that structure could change or new things added or descriptions changed? Then we'd have to remember to update them in these docs as well instead of just having people visit the one source of truth

Reply:
Just re-read your comment about autogenerating these docs and I totally agree with that!

Reply:
> when that structure could change or new things added or descriptions changed

Because API structure changes aren't automatically reflected here. We'd have to add those fields to the provider as well.